src/ng/directive/ngOptions.js

```diff
@@ -325,12 +325,28 @@ var ngOptionsDirective = ['$compile', '$parse', function($compile, $parse) {
         // The option values were already computed in the `getWatchables` fn,
         // which must have been called to trigger `getOptions`
         var optionValues = valuesFn(scope) || [];
+        var optionValuesKeys;
 
-        var keys = Object.keys(optionValues);
-        keys.forEach(function getOption(key) {
+
+        if (!keyName && isArrayLike(optionValues)) {
```

Comment:
Oh you mean this prevents `(key, value) in foos` from working? I don't think there's much point in that either